datastore/sqlite.nim

```diff
@@ -136,9 +142,26 @@ template journalModePragmaStmt*(env: SQLite): RawStmtPtr =
     s.dispose
     return failure $sqlite3_errstr(x)
 
-  if (let x = $sqlite3_column_text(s, 0).cstring; x != "memory" and x != "wal"):
+  let
+    x = sqlite3_column_text(s, 0).cstring
```

Comment:
Use sqlite3_get_text_not_null here too (see below)

Reply:
I've tentatively introduced `sqlite3_column_text_not_null` in `datastore/sqlite.nim` per b8e3b22f05c9a943370f4b19db66a0f65f32fd4f.

This naturally raises some questions re: earlier discussions, mainly https://github.com/status-im/nim-datastore/pull/17#discussion_r920319710:

If `sqlite3_column_text_not_null()` were to not raise a Defect for an OOM error, and instead we propagate the failure via `return failure`, then measures to handle `Result[T, E]` will naturally spread outward, e.g into `idCol` (hashtag  `#life-with-monads`).

If we're okay with `sqlite3_column_text_not_null()` raising a Defect for OOM error, then let's go with it. Otherwise I suggest dropping commit b8e3b22f05c9a943370f4b19db66a0f65f32fd4f.